test/tests/functional/pbs_snapshot_unittest.py

```diff
@@ -457,6 +457,79 @@ def test_obfuscate_resv_user_groups(self):
             self.assertFalse(str(TEST_USER1) in all_content)
             self.assertFalse(str(TSTGRP0) in all_content)
 
+    def test_obfuscate_acct_bad(self):
+        """
+        Test that pbs_snapshot --obfuscate can work with bad accounting records
+        """
+        if os.getuid() != 0:
+            self.skipTest("Test need to run as root")
+
+        if self.pbs_snapshot_path is None:
+            self.skip_test("pbs_snapshot not found")
+
+        # Delete all existing accounting logs
+        acct_logpath = os.path.join(self.server.pbs_conf["PBS_HOME"],
+                                    "server_priv", "accounting")
+        self.du.rm(path=os.path.join(acct_logpath, "*"), force=True,
+                   as_script=True)
+        ret = os.listdir(acct_logpath)
+        self.assertEqual(len(ret), 0)
+        self.server.pi.restart()
+
+        # Make sure that the restart generated a new accounting log
+        # Let's submit a job to generate a new accounting log
+        j = Job(TEST_USER)
+        j.set_sleep_time(1)
+        jid = self.server.submit(j)
+
+        # Check that the accounting E record was generated
+        self.server.accounting_match("E;%s;*" % jid, regexp=True)
```

Comment:
Your regex is wrong.  You are checking for zero or more semicolons.  You actually don't need to use regex.  log matching does substring matches.